### ISSUE TITLE
Missing d.ts files in fable-core 0.7.5-alpha.4

### ISSUE THREAD

**Issue by fdcastel** [fable-compiler/Fable#489]
### Description

The latest published `fable-core 0.7.5-alpha.4` doesn't contain `.d.ts` declaration files 😢.

Bad dog! No TypeScript for you! 

😜

**Comment by alfonsogarciacaro** [comment]
Ups, my bad 😅 I forgot to add `"declaration": "true"` to tsconfig.json. I did it and republished the package :+1: This is not without side effects though, I've changed all the previous static methods to module functions to improve tree shaking, but this means [I have to expose the private fields](https://github.com/fable-compiler/Fable/blob/aee38f2c4ec7af7b3fe9cdc3944baf6ffbea85c9/src/fable/Fable.Core/ts/Set.ts#L664-L665) of `Map` and `Set`, and because of that I also have to export `MapTree` and `SetTree` which should be private to the module.

At least it seems the `internal` modifier is coming to JS so that should fix the issue.
